tokio/src/macros/select.rs

```diff
@@ -0,0 +1,868 @@
+/// Wait on multiple concurrent branches, returning when the **first** branch
+/// completes, cancelling the remaining branches.
+///
+/// The `select` macro accepts one or more branches with the following pattern:
+///
+/// ```text
+/// <pattern> = <async expression> (, if <condition>)? => <handler>,
```

Comment:
minor nit: maybe we should refer to these as "preconditions" in the docs, rather than as "conditions"? just to reinforce that they're evaluated prior to polling.